### PR TITLE
suave-geth release checklist

### PR DESCRIPTION
**Pull request by metachris**

### CODE REVIEW

File: suave/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+- [ ] Pick the version (i.e. `v0.2.0`)
+- [ ] Tag new version (`git tag -s v0.2.0`)
+- [ ] Push tag to Github, CI will create a draft release
+- [ ] Edit the draft release to prepare nice release notes
+- [ ] Publish the release (note: this will send an email to subscribers on Github)
+
+### Test the published release
+
+- [ ] Download the release binary: `curl -L https://suaveup.flashbots.net | bash` and check the version with `suave-geth version`
```
Comment (ferranbt) [inline suave/release-checklist.md:32]: TODO: This does not show the version of suave-geth but the version from the geth fork.

Comment (metachris) [reply suave/release-checklist.md:32]: hehe, yeah that's a good point! definitely need to add this, as it's critical information for checking and debugging

Comment (metachris) [reply suave/release-checklist.md:32]: it's defined here: https://github.com/flashbots/suave-geth/blob/release-checklist/params/version.go#L23-L29

i'm assuming we'll just override this version number with the suave-geth one?

Comment (metachris) [reply suave/release-checklist.md:34]: added to publish steps

Comment (metachris) [reply suave/release-checklist.md:34]: updated

Comment (metachris) [reply suave/release-checklist.md:34]:
```bash
./build/bin/suave-geth version

Suave-Geth
Version: 0.1.1-dev
Git Commit: bbb9d5ee4f6fba64e31231092858484303aa82d0
Git Commit Date: 20240110
Architecture: arm64
Go Version: go1.21.0
Operating System: darwin
GOPATH=/Users/metachris/go
GOROOT=
```

**Review by ferranbt — APPROVED** [review]

**Pull request merged**